misc/systemd/ipfs.service

```diff
@@ -14,8 +28,11 @@ After=network.target
 Type=notify
 User=ipfs
 Group=ipfs
+TimeoutStartSec=15min
+TimeoutStopSec=15min
+TimeoutAbortSec=15min
 WorkingDirectory=/var/lib/ipfs
-ExecStart=/usr/bin/ipfs daemon --init --migrate
```

Comment:
Perhaps we need a separate service to run migrations? I agree that running migrations by default might get us to ugly places...

Reply:
I'm looking into options, but usually there's a script which is run by the package manager which either converts the database or returns immediately if there's no need to upgrade.

Here's [such a script](https://git.archlinux.org/svntogit/packages.git/tree/trunk/mariadb.install?h=packages/mariadb) as an example for an Arch package (in this case the service can start with the old database format):

Reply:
My issue with a separate service is, that it would be potentially called on the next boot, after an update.

@Stebalien [expressed](https://github.com/ipfs/go-ipfs/pull/7259#discussion_r418184621) that he likes the idea that IPFS can serve files on boot for booting services. So if a service would convert the database while the machine is booting, there's the potential that the boot of the machine will halt until the database conversion is completed.

I think the right way is to either hint the user to do a conversion, with a message from the package manager or have the package manager call a script to convert the database while applying the update - depending on the philosophy of the distribution.

Reply:
The right solution here is to:

1. Pass the `--migrate` flag.
2. Send `EXTEND_TIMEOUT_USEC=...` to the systemd notify socket repeatedly on start when migrating.

Reply:
Issue: https://github.com/ipfs/go-ipfs/issues/7269

Reply:
This sounds like a much better solution than mine. Thanks for looking into this!

I would still like to remove the --migration flag until #7269 is implemented (which can then add --migrate) back to the service. This way the service file can be used without risking that the conversation might be interrupted by systemd.

Reply:
Either way will break things:

* With the --migrate flag and with no extended timeout, we'll fail half-way through the migration if it happens to be a long one (the last few have been short).
* Without the --migrate flag, it'll _always_ fail to start.

We can make sure to fix #7269 before releasing a long-running migration.

Given this, I strongly disagree. Let's leave the --migrate flag.

Reply:
fixed